examples/migraphx/custom_op_miopen_kernel/custom_op_miopen_kernel.cpp

```diff
@@ -0,0 +1,173 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015-2022 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#include <algorithm>
+#include <hip/hip_runtime.h>
+#include <migraphx/migraphx.h>
+#include <miopen/miopen.h>
+#include <migraphx/migraphx.hpp> // MIGraphX's C++ API
+#include <numeric>
+#include <stdexcept>
+
+#define MIGRAPHX_MIOPEN_ASSERT(x) (assert((x) == miopenStatusSuccess))
+#define MIGRAPHX_HIP_ASSERT(x) (assert((x) == hipSuccess))
+
+inline miopenTensorDescriptor_t make_miopen_tensor(const migraphx::shape& s, bool pack = false)
+{
+    // TODO: normalize_standard shape for scalar
+    miopenTensorDescriptor_t t;
+    MIGRAPHX_MIOPEN_ASSERT(miopenCreateTensorDescriptor(&t));
+    // Convert to ints
+    auto s_lens = s.lengths();
+    std::vector<int> lens(s_lens.begin(), s_lens.end());
+    auto s_strides = s.strides();
+    std::vector<int> strides(s_strides.begin(), s_strides.end());
+    miopenDataType_t d;
+    if(s.type() == migraphx_shape_float_type)
+        d = miopenFloat;
+    else if(s.type() == migraphx_shape_half_type)
+        d = miopenHalf;
+    else if(s.type() == migraphx_shape_int32_type)
+        d = miopenInt32;
+    else if(s.type() == migraphx_shape_int8_type)
+    {
+        if(pack)
+        {
+            // update the lens and corresponding strides
+            d          = miopenInt8x4;
+            lens[1]    = ((lens[1] + 3) / 4) * 4;
+            strides[0] = strides[1] * lens[1];
+        }
+        else
+        {
+            d = miopenInt8;
+        }
+    }
+    else
+    {
+        throw("MAKE_TENSOR: unsupported type");
+    }
+    miopenSetTensorDescriptor(t, d, s_lens.size(), lens.data(), strides.data());
+    return t;
+}
+
+inline auto make_miopen_handle(migraphx::context& ctx)
+{
+    MIGRAPHX_HIP_ASSERT(hipSetDevice(0));
+    auto* stream = ctx.get_queue<hipStream_t>();
```

Comment:
Can we handle the case if get_queue () fails or returns null?

Reply:
Type check needs to happen inside that `get_queue<>` call. It is part of TODO. For now, we are only focusing on AMD GPU so, we are assuming, get_queue with hipStream will always be valid.  In order to do type check, it requires additional piece of work which on the TODO list.

Reply:
Understood, just wanted to ask. That makes sense though.